src/ddmd/expressionsem.d

```diff
@@ -8771,6 +8808,100 @@ Expression binSemanticProp(BinExp e, Scope* sc)
     return null;
 }
 
+/********************************************************************************
+ * Helper function to consolidate common code for visit(BinAssignExp)
+ * visit(PowAssignExp), and visit(CatAssignExp).  It rewrites expressions
+ * of the form, for example, `s.x += 1` to `s.x(s.x() += 1)` if s.x is
+ * a getter and setter `@property`.
+ * Params:
+ *      e = the expression to rewrite
+ *      sc = the semantic scope
+ * Returns:
+ *      the rewritten expression if the procedure succeeds, an `ErrorExp` if the
+ *      and error is encountered, or `null` if `e.e1` is not a `DotIdExp`.
+ */
+Expression binAssignRewriteProp(BinAssignExp e, Scope* sc)
+{
+    // https://issues.dlang.org/show_bug.cgi?id=8006
+
+    if (e.e1.op == TOKdotid)
```

Comment:
Maybe its better to complete semantic on `e1` first, then check that `e1.op == TOKcall` and `ce.e1.type.isproperty`.